File: rb101_rb109/easy8/double_char1.rb
=begin
input: a string
output: return a new string, every char is doubled
=end

# first attempt
def repeater(str)
  str.chars.map {|char| char * 2}.join
end  


# another solution using each_char
=begin
def repeater(str)
  result = ""
  str.each_char do |char|
    result << char * 2
  end
  result
end  
=end

puts repeater('Hello') 
puts repeater("Good job!") 
p repeater('')

File: js210/easy5/double_char1.js
"use strict";

function repeater(str) {
  let doubledStr = '';

  for (let idx = 0; idx < str.length; idx += 1) {
    doubledStr += (str[idx] + str[idx]);
  }
  return doubledStr;
}

console.log(repeater('Hello'));
console.log(repeater('Good job!'));
console.log(repeater(''));